### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by junpenglao**
* use new sampling api (advi init, etc)
* unify style

### CODE REVIEW

**Comment by junpenglao** [comment]
Following the discussion in #2236

File: pymc3/examples/baseball.py [hunk 1]
```diff
@@ -5,33 +5,34 @@
...
-    kappa = BoundedKappa( 'kappa', alpha=1.0001, m=1.5 )
-    thetas = pm.Beta( 'thetas', alpha=phi*kappa, beta=(1.0-phi)*kappa, shape=N )
-    ys = pm.Binomial( 'ys', n=atBats, p=thetas, observed=hits )
-
-def run( n=100000 ):
+def build_model():
+    data = np.loadtxt(pm.get_data('efron-morris-75-data.tsv'), delimiter="\t", 
+                      skiprows=1, usecols=(2,3))
+    
+    atBats = pm.floatX(data[:,0])
```
Comment (fonnesbeck) [inline pymc3/examples/baseball.py]: Avoid camelCase

File: pymc3/examples/baseball.py [hunk 2]
```diff
@@ -5,33 +5,34 @@
...
+        thetas = pm.Beta('thetas', alpha=phi*kappa, beta=(1.0-phi)*kappa, shape=N)
+        ys = pm.Binomial('ys', n=atBats, p=thetas, observed=hits)
+    return model
+
+def run(n=2000):
+    model = build_model()
     with model:
         # initialize NUTS() with ADVI under the hood
-        trace = pm.sample( n )
+        trace = pm.sample(n, nuts_kwargs={'target_accept':.99})
```
Comment (fonnesbeck) [inline pymc3/examples/baseball.py:32]: Wow, does it require this to converge?

Comment (junpenglao) [reply pymc3/examples/baseball.py:32]: it's fine without, but there is a divergent warning.

Comment (fonnesbeck) [reply pymc3/examples/baseball.py:32]: No problem; just surprised.

**Comment by junpenglao** [comment]
Remain to be done:
1. Better documentation: some of the examples need some background information (ref, etc). However, it is a bit difficult for me to do as I dont know the source either.
2. ~~`garch_example.py` is still broken as the changes in `Bound` method. I will do a separate PR with the fix in `Bound` also.~~ fixed in #2277 by @aseyboldt

**Comment by junpenglao** [comment]
Trying to fix `garch_example.py` and `Bound` and running into the same issue as #1491. Are there any clues from the last time this was fixed? @ColCarroll

**Comment by junpenglao** [comment]
Before Merging: @springcoil could you please test it in GPU again #2033?

File: pymc3/distributions/distribution.py [hunk 3]
```diff
@@ -476,8 +476,14 @@ class Bound(object):
 
     Example
     -------
-    boundedNormal = pymc3.Bound(pymc3.Normal, lower=0.0)
-    par = boundedNormal(mu=0.0, sd=1.0, testval=1.0)
+    # In general Bounded distribution need to be defined befor model context
```
Comment (junpenglao) [inline pymc3/distributions/distribution.py]: @fonnesbeck is this accurate?

Comment (fonnesbeck) [reply pymc3/distributions/distribution.py]: I've defined them inside a context before. I don't think it matters (it shouldn't). 

Also, before has an "e" at the end.

**Comment by springcoil** [comment]
I'll have a look this weekend maybe (swamped with work before then). Looks good work though I'll check it on a GPU though.

**Pull request merged**